File: creational/abstract-factory-method/src/main/java/GCPResourceFactory.java
import instances.GCPCloudEngine;
import instances.Instance;
import storages.GCPStorage;
import storages.Storage;

public class GCPResourceFactory implements ResourceCreatorFactory {
    @Override
    public Instance createInstance(Instance.CAPACITY capacity) {
        return new GCPCloudEngine(capacity);
    }

    @Override
    public Storage createStorage(int size) {
        return new GCPStorage(size);
    }
}

File: python/creational/abstract-factory/factory/GCPResourceFactory.py
from ResourceFactory import ResourceFactory
from instance.GoogleCloudInstance import GoogleCloudInstance
from storage.GCPStorage import GCPStorage


class GCPResourceFactory(ResourceFactory):

    def create_instance(self):
        return GoogleCloudInstance()

    def create_storage(self, size):
        return GCPStorage(size)
